api/dp-openapi.yaml

```diff
@@ -678,6 +678,315 @@ paths:
         default:
           $ref: '#/components/responses/UnexpectedError'
 
+  /projects/{projectid}/checkout:
+    put:
+      tags:
+      - project
+      description: Check out some pages for proofreading
+      parameters:
+      - $ref: '#/components/parameters/projectid'
+      - $ref: '#/components/parameters/projectstate'
+      - name: npage
+        in: query
+        description: number of pages to checkout
+        required: false
+        schema:
+          type: integer
+          default: 1
+      responses:
+        200:
+          description: result status
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/checkout_status'
+              example:
+                status: 303
+                reason: You have reached your quota of pages from 'Beginners Only' projects in this round.
+                npages: 2
+        '400':
+          $ref: '#/components/responses/InvalidValue'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '429':
+          $ref: '#/components/responses/RateLimitExceeded'
+        default:
+          $ref: '#/components/responses/UnexpectedError'
+
+  /projects/{projectid}/mentorcheckout:
+    put:
+      tags:
+      - project
+      description: Finds the 'oldest' mentee and checks out all his/her pages.
+      parameters:
+      - $ref: '#/components/parameters/projectid'
+      - $ref: '#/components/parameters/projectstate'
+      responses:
+        200:
+          description: no content
+        '400':
+          $ref: '#/components/responses/InvalidValue'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '429':
+          $ref: '#/components/responses/RateLimitExceeded'
+        default:
+          $ref: '#/components/responses/UnexpectedError'
+
+  /projects/{projectid}/returnpages:
```

Comment:
This feels like an unnecessary abstraction and should be left up to the client to return each page individually.

Reply:
> This feels like an unnecessary abstraction and should be left up to the client to return each page individually.

Bear in mind that I know nothing about designing an API, but it makes a lot of sense to me that the API be simple and straightforward, and that the software using the API determine how it's implemented.

Reply:
> This feels like an unnecessary abstraction and should be left up to the client to return each page individually.

This was actually requested in an earlier demonstration of a method of implementing a mentor's proofreading interface.

Reply:
> > This feels like an unnecessary abstraction and should be left up to the client to return each page individually.
> 
> This was actually requested in an earlier demonstration of a method of implementing a mentor's proofreading interface.

Come to think of it, I was probably the one who pointed out that if a proofer has several pages to return, they might not bother. Of course that's true now, also. Does the "return block" have to be specified in the API?

Reply:
> Does the "return block" have to be specified in the API?

Not necessarily. But it would make life easier. Otherwise the client software would have to get each checked-out page and return it until there were no more left. So returning 5 pages would need 11 api calls, 5 returning the text payload, and the last reporting there were no more pages checked out.

(In the demo referred to this function is called from the project page rather than the Proofreading interface).